sycl/include/CL/sycl/ordered_queue.hpp

```diff
@@ -19,12 +19,15 @@
 
 #ifdef __has_cpp_attribute
   #if __has_cpp_attribute(deprecated)
-    #define __SYCL_DEPRECATED__ [[deprecated("Replaced by in_order queue property")]]
+    #if __cplusplus >= 201402
+      #define __SYCL_DEPRECATED__                                              \
+        [[deprecated("Replaced by in_order queue property")]]
+    #else
+      #define __SYCL_DEPRECATED__ __attribute__                                \
+        ((deprecated("Replaced by in_order queue property")))
+    #endif
   #endif
 #endif
-#ifndef __SYCL_DEPRECATED__
-  #define __SYCL_DEPRECATED__
-#endif
```

Comment:
Please, return this block as well. Otherwise, we might end-up with using undefined macro.

Reply:
Significantly changed the patch, please review again.